pkg/utils/kubernetes/resources/secrets.go

```diff
@@ -42,28 +43,32 @@ func WithAnnotation[T client.Object](k, v string) func(d T) {
 	}
 }
 
-type controlPlaneOrDataPlane interface {
-	*operatorv1beta1.ControlPlane | *operatorv1beta1.DataPlane
+type controlPlaneDataPlaneOrKonnectExtension interface {
```

Comment:
Can we export this type and move it somewhere where it can be shared and use in `controller/pkg/secrets/cert.go` where we redefine it?

Reply:
Why should we? That interface is used only here. I'd lean toward leaving it here and maybe moving it somewhere else when we'll need it somewhere else.

Reply:
All I'm saying is that https://github.com/Kong/gateway-operator/pull/1304/files#diff-465e5135e663acc85bb2b22849039cb833ae715282354383ac12ee3d638b67feR180 is also defining a type constraint based on these 3 types. I'm OK if we leave it as is but I think this could be simplified by reusing the type constraint.

Reply:
I apologize, I didn't catch the point. It makes sense, here it is: 6ac83e72a66bc513afa5e4287a3d3ec0d4f9f5d9